.github/workflows/on-pullrequest.yml

```diff
@@ -16,6 +21,67 @@ defaults:
     working-directory: Meadowlark-js
 
 jobs:
+  scan-actions:
+    name: Scan Actions
+    uses: ed-fi-alliance-oss/ed-fi-actions/.github/workflows/repository-scanner.yml@latest
+
+  analyze-code:
+    if: github.event_name == 'pull_request'
+    name: Analyze Code Dependencies
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b # v3.0.2
```

Comment:
Can you look to see if there are newer & better versions available for

* checkout
* dependency-review-action
* setup-node
* actions-cache
* github/codeql-action